src/main/scala/org/apache/spark/sql/delta/hooks/GenerateManifest.scala

```diff
@@ -16,40 +16,47 @@
 
 package org.apache.spark.sql.delta.hooks
 
-import org.apache.spark.sql.delta._
-import org.apache.spark.sql.delta.actions._
-import org.apache.spark.sql.delta.metering.DeltaLogging
-import org.apache.spark.sql.delta.storage.LogStore
-import org.apache.spark.sql.delta.util.DeltaFileOperations
 import org.apache.hadoop.fs.Path
-
-import org.apache.spark.SparkEnv
 import org.apache.spark.sql._
 import org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute
 import org.apache.spark.sql.catalyst.catalog.ExternalCatalogUtils
 import org.apache.spark.sql.catalyst.expressions.{Attribute, Cast, Concat, Expression, Literal, ScalaUDF}
+import org.apache.spark.sql.delta._
+import org.apache.spark.sql.delta.actions._
+import org.apache.spark.sql.delta.metering.DeltaLogging
+import org.apache.spark.sql.delta.storage.LogStore
 import org.apache.spark.sql.execution.datasources.InMemoryFileIndex
 import org.apache.spark.sql.functions.col
 import org.apache.spark.sql.types.StringType
 import org.apache.spark.util.SerializableConfiguration
 
-/**
- * Post commit hook to generate hive-style manifests for Delta table. This is useful for
- * compatibility with Presto / Athena.
- */
-object GenerateSymlinkManifest extends GenerateSymlinkManifestImpl
```

Comment:
What do you think about having `GenerateManifestImpl` the base class for both Symlink and Json.

Reply:
Fine to me, 
Actually, I thought you would like some structure like below as incrementChangeCapture on what partitions added/ removed is quite usefull, but would need a bigger change which is better for you guys to do in a bigger refactor.
![image](https://user-images.githubusercontent.com/33287576/78629694-5996bf80-78ca-11ea-961c-49c8d660017b.png)